src/ngraph/op/fused/group_conv.cpp

```diff
@@ -95,20 +120,33 @@ void op::GroupConvolution::post_validate_and_infer_types()
 
 Shape op::GroupConvolution::get_weights_dimensions() const
 {
+    auto data_shape = get_input_shape(0);
+    auto weights_shape = get_input_shape(1);
+    // check if weights already includes groups
+    if (has_groups_in_filters_shape())
+    {
+        return weights_shape;
+    }
     // reshape weights into 5d tensors that includes groups
     const size_t OC = 0;
     const size_t OC_IN_OUTPUT = 1;
     const size_t IC = 1;
-    Shape weights_shape_groups{get_input_shape(1)};
+    Shape weights_shape_groups{weights_shape};
     // adjust output and channel given a number of groups
 
     weights_shape_groups.at(OC) = get_shape().at(OC_IN_OUTPUT) / get_groups();
-    weights_shape_groups.at(IC) = get_input_shape(0).at(IC) / get_groups();
+    weights_shape_groups.at(IC) = data_shape.at(IC) / get_groups();
     // push_front the number of groups
     weights_shape_groups.insert(weights_shape_groups.begin(), get_groups());
     return weights_shape_groups;
 }
 
+size_t ngraph::op::GroupConvolution::get_groups() const
+{
+    NODE_VALIDATION_CHECK(this, m_groups.is_static(), "Number of groups must be static");
```

Comment:
Probably better to say `get_groups can only be called if the number of groups is static`, or something like that.

Reply:
Sounds good, done.